### ISSUE TITLE
Players getting timed out

### ISSUE THREAD

**Issue by Khrystyian** [PaperMC/Velocity#1441]
### Expected Behavior

Players should not be kicked randomly.

### Actual Behavior

[00:04:57] [Netty epoll Worker #3/INFO] [com.velocitypowered.proxy.connection.client.ConnectedPlayer]: [connected player] Enderzombie44 (/163.117.104.252:50180) has disconnected: An internal error occurred in your connection.
[00:04:57] [Netty epoll Worker #3/ERROR] [com.velocitypowered.proxy.connection.MinecraftConnection]: [connected player] Enderzombie44 (/163.117.104.252:50180): read timed out

Players are getting kicked randomly.

### Steps to Reproduce

https://pastebin.com/dnpJYEtG

### Plugin List

There are no plugins currently installed.

### Velocity Version

[11:49:33 INFO]: Velocity 3.3.0-SNAPSHOT (git-0cd069ec-b431)
[11:49:33 INFO]: Copyright 2018-2024 Velocity Contributors. Velocity is licensed under the terms of the GNU General Public License v3.
[11:49:33 INFO]: velocitypowered.com - GitHub

**Comment by bolda500** [comment]
I have the same problem, and I assume I don't have the same plugins as the guy above. Sometimes the connection is broken a dozen or so seconds after connecting and sometimes after a longer time. On average, a player is kicked out every few minutes, with 60-70 online.

Below is the error log.

ERROR censored-nick (/77.222.*.*:22421): exception encountered in com.velocitypowered.proxy.connection.client.ClientPlaySessionHandler@31984e67
10.10 23:12:17 [Server] INFO io.netty.channel.unix.Errors$NativeIoException: recvAddress(..) failed: Connection reset by peer

**Comment by Stis780** [comment]
Yes sir! We have exact the same problem, a bunch of people gets disconnected, usually around peak, probably high load or some kind of race condition? it's still bad they get kicked off the proxy

**Comment by electronicboy** [comment]
connection reset by peer means that the connection was closed externally to the proxy;
read timed out means that the proxy didn't get any packets in 30 seconds on that connection;

Diagnosing these types of things is generally hard, especially with 0 information, large amounts of disconnects at once would generally be networking issues or something, like a plugin, stalling out a network pipeline thread. without figuring out a way to throw a lot of monitoring into your specific setups, it's generally not tenable for us to advise much here

**Comment by Stis780** [comment]
Update netty? I know it's hard to do this, is there any way we can help? How?

**Comment by electronicboy** [comment]
updating netty to the latest 4.1 release would be trivial, but there have been no updates that would be relevant to connections being closed externally

**Comment by bolda500** [comment]
sorry if I misled you, I changed the network_compression_threshold from 512 to 64 and the problem disappeared, I don't know why it was set like that but now it's ok, I also changed the version of Velocity to the last build of 3.4.0 but I don't know if it has anything to do with it. I will try to monitor closely whether the problem still occurs. Thank you for your time.

**Comment by Stis780** [comment]
On velocity? Isn't default there 256?

**Comment by Stis780** [comment]
BTW Also not sure if this is intended but https://papermc.io/downloads/velocity still defaults to 3.3.0

**Comment by Timongcraft** [comment]
> BTW Also not sure if this is intended but https://papermc.io/downloads/velocity still defaults to 3.3.0

Already reported:
https://github.com/PaperMC/docs/issues/487
(reference: https://discord.com/channels/289587909051416579/947676006964162611/1293570377288650793)

**Comment by bolda500** [comment]
> On velocity? Isn't default there 256?

of course it is, but it's possible that I changed something by mistake.

**Comment by Stis780** [comment]
@bolda500 did this improve things in any meaningful way? I was already using 256 and the issue still occurs...

**Comment by bolda500** [comment]
> [@bolda500](https://github.com/bolda500) did this improve things in any meaningful way? I was already using 256 and the issue still occurs...

After changing the value to 64, or I guess 128 would also be good, the problem disappeared. I also changed the **connection-timeout** value from 5000 to 7000.

**Comment by iae-st** [comment]
Seems like our server started to run into this too. Every once in a while velocity has a read timeout. We tried changing the network_compression_threshold to 64 and 128. We also attempted to change the connection-timeout value. Nothing seemed to fix it. Would be great if anyone has found a solution for this error. (We also tried it with no plugins at all besides Velocity and it was still occurring)

**Comment by bolda500** [comment]
> Seems like our server started to run into this too. Every once in a while velocity has a read timeout. We tried changing the network_compression_threshold to 64 and 128. We also attempted to change the connection-timeout value. Nothing seemed to fix it. Would be great if anyone has found a solution for this error. (We also tried it with no plugins at all besides Velocity and it was still occurring)

What Java GC engine you use?

**Comment by iae-st** [comment]
> > Seems like our server started to run into this too. Every once in a while velocity has a read timeout. We tried changing the network_compression_threshold to 64 and 128. We also attempted to change the connection-timeout value. Nothing seemed to fix it. Would be great if anyone has found a solution for this error. (We also tried it with no plugins at all besides Velocity and it was still occurring)
> 
> What Java GC engine you use?

We're currently using G1GC

**Comment by Hackerjef** [comment]
@bolda500 @KhrysAK47  by any chance are either of you using Proxy protocol v2?

**Comment by bridgelol** [comment]
Also experiencing this

**Comment by loqqin** [comment]
also experiencing this
what hosting service do you use? I use hetzner

**Comment by iae-st** [comment]
> also experiencing this what hosting service do you use? I use hetzner

We were using Hetzner at the time, we swapped but also ended up finding a different issue from another plugin which was causing disconnects too. Now that we removed that plugin and swapped away from Hetzner we have barely experienced any issues.

**Comment by 0xF5T9** [comment]
Our server is also experiencing this. Has there any update on this matter?

**Comment by tirangaming** [comment]
I am facing the same issue. Any updates?
